nailgun/entities.py

```diff
@@ -8132,3 +8132,149 @@ def __init__(self, server_config=None, **kwargs):
+                    "actions.remote_execution.run_host_job_katello_group_install_succeeded",
+                    "actions.remote_execution.run_host_job_katello_group_remove_succeeded",
+                    "actions.remote_execution.run_host_job_katello_group_update_succeeded",
+                    "actions.remote_execution.run_host_job_katello_module_stream_action_succeeded",
+                    "actions.remote_execution.run_host_job_katello_package_install_succeeded",
+                    "actions.remote_execution.run_host_job_katello_package_remove_succeeded",
+                    "actions.remote_execution.run_host_job_katello_package_update_succeeded",
+                    "actions.remote_execution.run_host_job_katello_service_restart_succeeded",
+                    "actions.remote_execution.run_host_job_puppet_run_host_succeeded",
+                    "actions.remote_execution.run_host_job_rh_cloud_remediate_hosts_succeeded",
+                    "actions.remote_execution.run_host_job_succeeded",
+                    "build_entered",
+                    "build_exited",
+                    "content_view_created",
+                    "content_view_destroyed",
+                    "content_view_updated",
+                    "domain_created",
+                    "domain_destroyed",
+                    "domain_updated",
+                    "host_created",
+                    "host_destroyed",
+                    "host_updated",
+                    "hostgroup_created",
+                    "hostgroup_destroyed",
+                    "hostgroup_updated",
+                    "model_created",
+                    "model_destroyed",
+                    "model_updated",
+                    "status_changed",
+                    "subnet_created",
+                    "subnet_destroyed",
+                    "subnet_updated",
+                    "user_created",
+                    "user_destroyed",
+                    "user_updated",
+                ],
+            ),
+            'http_content_type': entity_fields.StringField(),
+            'http_method': entity_fields.StringField(
+                choices=[
+                    "POST",
+                    "GET",
+                    "PUT",
+                    "DELETE",
+                    "PATCH",
+                ]
+            ),
+            'password': entity_fields.StringField(),
+            'proxy_authorization': entity_fields.BooleanField(),
+            'target_url': entity_fields.URLField(required=True),
+            'user': entity_fields.StringField(),
+            'verify_ssl': entity_fields.BooleanField(),
+            'webhook_template': entity_fields.OneToOneField(WebhookTemplate),
+        }
+        self._meta = {
+            'api_path': 'api/webhooks',
+        }
+        super().__init__(server_config, **kwargs)
+
+    def create(self, create_missing=None):
+        """
+        Create WebhookTemplate properly when creating a parent Webhook instance
```

Comment:
can you elaborate on why this entity deviates from the normal pattern?

Reply:
I see he is returning `webhook_template` as WebhookTemplate class object but that should have been already returning.

I think you are looking for all the WebHookTemplate entity attributes to return after creating the WebHook entity, whereas, its by default might be returning only id, name etc but not all!

you could explicit read evertime you need all attributes in tests / helpers:
```
>>> whook = entities.Webhook(name='whook1', webhook_template=entities.WebhookTemplate(id=1).read()).create()
>>> whook_template = whook.webhook_template
entities.WebhookTemplate(id=1)
>>> whook.read()
entities.WebhookTemplate(id=1, name='adff', template='aasas3', snippet=True, ........)
```